thorlcr/master/thmastermain.cpp

```diff
@@ -314,74 +326,73 @@ class CRegistryServer : public CSimpleInterface
                 StringBuffer workerPodName, workerContainerName;
                 if (NotFound == slaveNum)
                 {
-                    connectedSlaves.append(sender.getLink());
-                    slaveNum = connectedSlaves.ordinality();
                     if (isContainerized())
                     {
                         msg.read(workerPodName);
                         msg.read(workerContainerName);
-                        addConnectedWorkerPod(workerPodName, workerContainerName); // NB: these are added in worker # order
                     }
+                    connectedWorkers.emplace_back(workerEPStr.str(), workerPodName, workerContainerName);
+                    PROGLOG("Slave connected from %s", workerEPStr.str());
                 }
                 else
                 {
                     unsigned pos = slaveNum - 1; // NB: slaveNum is 1 based
-                    while (connectedSlaves.ordinality() < pos)
-                        connectedSlaves.append(nullptr);
-                    if (connectedSlaves.ordinality() == pos)
-                        connectedSlaves.append(sender.getLink());
+                    while (connectedWorkers.size() < pos)
+                        connectedWorkers.emplace_back(std::string(), std::string(), std::string());
```

Comment:
trivial: following may be clearer?
 
connectedWorkers.emplace_back()

Reply:
didn't notice could do that, but yes, but have also changed it to use struct vs tuple now.